OmgvaPOS/src/ItemVariationManagement/Validators/ItemVariationValidator.cs

```diff
@@ -29,4 +29,34 @@ public static void EnoughInventoryQuantity(ItemVariation itemVariation, int amou
         if (itemVariation.InventoryQuantity < amount)
             throw new ValidationException($"Not enough {itemVariation.Name} left. Inventory has: {itemVariation.InventoryQuantity}, requested: {amount}.");
     }
+    
+    public static void ValidateNoItemVariationsFromSameGroup(List<ItemVariation>? itemVariations) {
+        if (itemVariations == null || itemVariations.Count < 2)
+            return;
+        
+        var variationGroups = itemVariations
+            .Select(oiv => oiv.ItemVariationGroup)
+            .GroupBy(group => group)
+            .ToList();
+
+        // If any group appears more than once, throw an exception
+        var duplicateGroups = variationGroups
+            .Where(g => g.Count() > 1)
+            .Select(g => g.Key)
+            .ToList();
+
+        if (duplicateGroups.Count != 0)
+        {
+            throw new ValidationException(
+                $"Multiple variations from the same group(s) are not allowed: {string.Join(", ", duplicateGroups)}"
+            );
+        }
+    }
+
+    public static void ValidateItemVariationBelongsToItem(ItemVariation itemVariation, Item item)
+    {
+        if (itemVariation.ItemId != item.Id)
+            throw new ValidationException($"Variation {itemVariation.Name} does not belong to item {item.Name}");
```

Comment:
This should probably throw not found.

Reply:
Not really, since we found the variation, however it does not belong to the item. 
For example 'Variation _extra sauce_ does not belong to item _Coffee with milk_" , because we don't just put ketchup in coffee

Reply:
Does it belong to the same business?

Reply:
ItemVariation does not have a direct BusinessId, but has a reference to Item. If item is from business then itemVaration belongs to the same business. Added validation that item is from same business as the order
![image](https://github.com/user-attachments/assets/7aa2b4c1-1540-4e66-bdd9-cf1665244d58)

Reply:
Still sceptical if we should inform that such item exists at all. Especially if it is from another business

Reply:
Changed to "Item not found in business"